python/src/equistore/operations/allclose.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+
+from equistore import TensorBlock, TensorMap
+
+from . import _dispatch
+from ._utils import _check_same_gradients, _check_same_keys
+
+
+def allclose(
+    tensor1: TensorMap, tensor2: TensorMap, rtol=1e-13, atol=1e-12, equal_nan=False
+) -> bool:
+    """Confront two :py:class:`TensorMap`.
+    It is True the two tensor have the same keys (even in different order)
+    and all the :py:class:`TensorBlock` have the same samples, components,
+    properties and
+    their values matrices must pass the numpy-like allclose test with rtol, atol.
+
+    If some :py:class:`TensorBlock` have gradients the gradient must have, as well, same
+    samples, components, properties and their data matrices must pass the numpy-like
+    allclose test with rtol, atol.
+
+    :param tensor1: (TensorMap): first :py:class:`TensorMap`.
+    :param tensor2: (TensorMap): second :py:class:`TensorMap`.
+    :param rtol: (float, optional): The relative tolerance parameter. Defaults: 1e-13.
+    :param atol: (float, optional): The relative tolerance parameter. Defaults: 1e-12.
+    :param equal_nan: (bool,optional): If True then two `NaN` will be considered equal.
+                               Defaults: False.
+    """
+    try:
+        _allclose_tensor(
+            tensor1=tensor1, tensor2=tensor2, rtol=rtol, atol=atol, equal_nan=equal_nan
+        )
+        return True
+    except:  # noqa
```

Comment:
```suggestion
    except ValueError:
```

Or we could introduce a new exception type